File: app/src/main/java/ru/michanic/mymot/Models/HtmlAdvertsAsyncRequest.java
package ru.michanic.mymot.Models;

import android.os.AsyncTask;
import android.util.Log;

import org.jsoup.Connection;
import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;

import java.io.IOException;

import ru.michanic.mymot.Enums.SourceType;
import ru.michanic.mymot.MyMotApplication;
import ru.michanic.mymot.Protocols.AsyncRequestCompleted;

public class HtmlAdvertsAsyncRequest extends AsyncTask<String, Void, ParseAdvertsResult> {

    public AsyncRequestCompleted delegate = null;
    private HtmlParser htmlParser = new HtmlParser();
    private SourceType sourceType;

    public HtmlAdvertsAsyncRequest(AsyncRequestCompleted asyncResponse, SourceType sourceType) {
        this.delegate = asyncResponse;
        this.sourceType = sourceType;
    }


    @Override
    protected ParseAdvertsResult doInBackground(String... arg) {
        String path = arg[0];
        Connection.Response response = MyMotApplication.networkService.getHtmlData(path);
        Document doc = null;
        try {
            doc = response.parse();
        } catch (IOException e) {
            e.printStackTrace();
        }
        ParseAdvertsResult parsedAdverts = htmlParser.parseAdverts(doc, sourceType);
        return parsedAdverts;
    }

    @Override
    protected void onPostExecute(ParseAdvertsResult result) {
        //super.onPostExecute(result);
        delegate.processFinish(result);
    }

}

File: app/src/main/java/ru/michanic/mymot/Kotlin/Models/HtmlAdvertsAsyncRequest.kt
package ru.michanic.mymot.Kotlin.Models

import android.os.AsyncTask
import org.jsoup.nodes.Document
import ru.michanic.mymot.Kotlin.Enums.SourceType
import ru.michanic.mymot.Kotlin.MyMotApplication
import ru.michanic.mymot.Kotlin.Protocols.AsyncRequestCompleted
import java.io.IOException

class HtmlAdvertsAsyncRequest(asyncResponse: AsyncRequestCompleted?, sourceType: SourceType) :
    AsyncTask<String?, Void?, ParseAdvertsResult>() {
    var delegate: AsyncRequestCompleted? = null
    private val htmlParser = HtmlParser()
    private val sourceType: SourceType

    init {
        delegate = asyncResponse
        this.sourceType = sourceType
    }

    override fun doInBackground(vararg arg: String?): ParseAdvertsResult {
        val path = arg[0]
        val response =
            MyMotApplication.networkService?.getHtmlData(path)
        var doc: Document? = null
        try {
            doc = response?.parse()
        } catch (e: IOException) {
            e.printStackTrace()
        }
        return htmlParser.parseAdverts(doc, sourceType)
    }

    override fun onPostExecute(result: ParseAdvertsResult) {
        delegate?.processFinish(result)
    }
}